clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -80,13 +81,21 @@
     /* Listener to be invoked when assignment changes */
     private ConsumerRebalanceListener listener;
 
+    private final Metrics metrics;
+
+    // The constructor is only intended to be used by unit tests.
     public SubscriptionState(OffsetResetStrategy defaultResetStrategy) {
```

Comment:
I am not sure it's worth keeping this constructor since it would be wrong to call it outside of tests. And there are only 5 usages and it's not too onerous to just pass an instance of `Metrics`.